### PR TITLE
MGMT-12837: Add Shielded VMs configuration to gcp provider types

### PR DESCRIPTION
**Pull request by eranco74**
Add support to Shielded VMs on GCP
Feature link: https://issues.redhat.com/browse/OCPBU-158

### CODE REVIEW

**Comment by JoelSpeed** [comment]
I'd like to see this feature contributed to the upstream cluster API provider for GCP before we merge this downstream
/hold

**Comment by eranco74** [comment]
> I'd like to see this feature contributed to the upstream cluster API provider for GCP before we merge this downstream /hold

https://github.com/kubernetes-sigs/cluster-api-provider-gcp/pull/777

**Comment by JoelSpeed** [comment]
API changes look ok, as this feature will be supported by the Machine API team, I'd like to see their QE and docs team ack this before we merge.

CC @jeana-redhat @sunzhaohua2 will you both have capacity to document and test this feature respectively within 4.12?

CC @makentenza can you please link the ocpcloud epic for this once you've created it so we can get the acks recorded there

**Comment by JoelSpeed** [comment]
Apologies, just to be clear, I meant 4.13, not 4.12, my bad

**Comment by jeana-redhat** [comment]
Should be no problem for 4.13, thanks!

File: machine/v1beta1/types_gcpprovider.go [hunk 1]
```diff
@@ -201,3 +235,29 @@ type GCPMachineProviderStatus struct {
...
+	// +optional
+	VirtualizedTrustedPlatformModule VirtualizedTrustedPlatformModulePolicy `json:"virtualizedTrustedPlatformModule,omitempty"`
+
+	// IntegrityMonitoring determines whether the instance should have integrity monitoring that verify the runtime boot integrity.
+	// Compares the most recent boot measurements to the integrity policy baseline and return
+	// a pair of pass/fail results depending on whether they match or not.
+	// If omitted, the platform chooses a default, which is subject to change over time, currently that default is Enabled.
+	// +kubebuilder:validation:Enum=Enabled;Disabled
+	// +optional
+	IntegrityMonitoring IntegrityMonitoringPolicy `json:"integrityMonitoring,omitempty"`
```
Comment (JoelSpeed) [inline machine/v1beta1/types_gcpprovider.go:262]: During testing, QE discovered a relationship between these two fields, this needs to be documented in the Godoc here before we merge @eranco74

**Comment by JoelSpeed** [comment]
What's the status of getting this feature implemented in the upstream GCP cluster API provider?

**Comment by JoelSpeed** [comment]
Ahh upstream is good
/hold cancel

**Comment by eranco74** [comment]
cc @miyadav, @jeana-redhat, @sferich888
Please add the qe-approved, docs-approved px-approved labels

**Comment by jeana-redhat** [comment]
LGTM for 4.13

/label docs-approved

**Pull request merged**